source/numerics/nonlinear.cc

```diff
@@ -0,0 +1,238 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2017 - 2022 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/numerics/nonlinear.h>
+
+#  include <deal.II/base/utilities.h>
+
+#  include <deal.II/lac/block_vector.h>
+#  include <deal.II/lac/la_parallel_block_vector.h>
+#  include <deal.II/lac/la_parallel_vector.h>
+#  ifdef DEAL_II_WITH_TRILINOS
+#    include <deal.II/lac/trilinos_parallel_block_vector.h>
+#    include <deal.II/lac/trilinos_vector.h>
+#  endif
+#  ifdef DEAL_II_WITH_PETSC
+#    include <deal.II/lac/petsc_block_vector.h>
+#    include <deal.II/lac/petsc_vector.h>
+#  endif
+
+#  include <iomanip>
+#  include <iostream>
+
+DEAL_II_NAMESPACE_OPEN
+
+
+  template <typename VectorType>
+  Nonlinear_Solver<VectorType>::AdditionalData_NL::AdditionalData_NL(
+    const SolverType       &solvertype,
+    const SolutionStrategy &strategy,
+    const unsigned int      maximum_non_linear_iterations,
+    const double            function_tolerance,
+    const double            step_tolerance,
+    const bool              no_init_setup,
+    const unsigned int      maximum_setup_calls,
+    const double            maximum_newton_step,
+    const double            dq_relative_error,
+    const unsigned int      maximum_beta_failures,
+    const unsigned int      anderson_subspace_size)
+    : solvertype(solvertype)
+    , strategy(strategy)
+    , maximum_non_linear_iterations(maximum_non_linear_iterations)
+    , function_tolerance(function_tolerance)
+    , step_tolerance(step_tolerance)
+    , no_init_setup(no_init_setup)
+    , maximum_setup_calls(maximum_setup_calls)
+    , maximum_newton_step(maximum_newton_step)
+    , dq_relative_error(dq_relative_error)
+    , maximum_beta_failures(maximum_beta_failures)
+    , anderson_subspace_size(anderson_subspace_size)
+  {}
+
+
+
+  template <typename VectorType>
+  void
+  Nonlinear_Solver<VectorType>::_NL::add_parameters(ParameterHandler &prm)
```

Comment:
What is ``_NL``?

Reply:
This .cc file shouldn't exist, I'll remove it in my next push.